src/fabric.c

```diff
@@ -62,12 +62,7 @@ pthread_mutex_t ini_lock = PTHREAD_MUTEX_INITIALIZER;
 
 static struct fi_filter prov_filter;
 
-struct fi_provider core_prov = {
-	.name = "core",
-	.version = 1,
-	.fi_version = FI_VERSION(FI_MAJOR_VERSION, FI_MINOR_VERSION)
-};
-
+extern struct fi_provider core_prov;
```

Comment:
it seems it is historicaly copied from source to source... I can add reference to core_prov into header, but I don't see good place for it, what about `fi.h`?